docs/java/java-project.md

```diff
@@ -79,6 +61,22 @@ If you have multiple subfolders that have source code for your project, you can
   <source src="/docs/java/java-project/multiple-source.mp4" type="video/mp4">
 </video>
 
+## Lightweight Mode
+
+VS Code for Java supports two modes, lightweight and standard. With lightweight mode, only source files and JDK are resolved by language server; with standard mode, imported dependencies are resolved and project is built by language server. Lightweight mode works best when you need a super quick-to-start and lightweight environment to work with your source files, for example, reading source code, navigating among source code and JDK, viewing outline and Javadoc, and detecting and fixing syntax errors. Also, code completion is supported within the scope of source files and JDK.
+
+Lightweight mode does neither resolve imported dependencies nor build project, so it does not support run, debugging, refactoring, linting or detecting semantic errors. For taking advantage of these features, you need switch your workspace from lightweight mode to standard mode.
+
+You can control which mode to launch with by configuring `java.server.launchMode` with below options:
+
+- `Hybrid` (default) - Firstly, a workspace is opened with lightweight mode. You will be asked whether to switch to standard mode if your workspace contains unresolved Java project. If you choose "Later", it will stay in lightweight mode. You can click the server mode icon on status bar to manually switch to standard mode.
+- `Standard` - A workspace is opened with standard mode.
+- `LightWeight` - A workspace is opened with lightweight mode. You can click the server mode icon on status bar to manually switch to standard mode.
+
+Status bar indicates which mode current workspace is in using different icons. ![rocket](images/java-project/lightweight-icon.PNG) workspace opened with lightweight mode; ![synchronisation](images/java-project/processing-icon.png) workspace being opened with standard mode; ![thumb up](images/java-project/standard-icon.PNG) workspace opened with standard mode. When in hybrid mode, clicking lightweight mode icon takes you to standard mode.
```

Comment:
`<i class="codicon codicon-rocket"></i>` - workspace opened with lightweight mode; `<i class="codicon codicon-sync"></i>` -workspace being opened with standard mode; `<i class="codicon codicon-thumbsup"></i>` - workspace opened with standard mode. Clicking lightweight mode icon takes you to standard mode.

The screenshot is not aligned with the right text. Maybe it's better to use the octicons directly. See icons page at https://code.visualstudio.com/api/references/icons-in-labels#icon-listing. Its markdown is https://github.com/microsoft/vscode-docs/edit/master/api/references/icons-in-labels.md

Not sure how to preview it in github page.

Reply:
Looks like the markdown i pasted above doesn't work, but if you put some HTML tags below to the markdown directly, it render well at vscode-docs websites.

\<div id="codicon-listing"\>
Status bar indicates which mode current workspace is in using different icons. \<i class="codicon codicon-rocket">\</i> - workspace opened with lightweight mode; \<i class="codicon codicon-sync">\</i> - workspace being opened with standard mode; \<i class="codicon codicon-thumbsup"></i> - workspace opened with standard mode. Clicking lightweight mode icon takes you to standard mode.
\</div>

The preview:
![image](https://user-images.githubusercontent.com/14052197/89002256-325b7680-d32f-11ea-9748-8e9faf681356.png)

Reply:
@gregvanl, I updated with Jinbo's suggestion using codicon. But, I can't validate since the validation requires CSS loaded. I am using VS Code for editing. Do you have any suggestion how to validate?

Reply:
Any reason you don't want to just copy the emoji inline in the text? It should work and there are several examples such as https://code.visualstudio.com/updates/v1_47#_stable-windows-arm-builds

Reply:
Another option would be to show screenshots of the icons in the Status bar

Reply:
What the Java extension itself used is codicons. Emoji is a little different. Using codicons here can keep the consistent with the extension.

Reply:
@yaojinyang Here is the vscode-codicons repo https://github.com/microsoft/vscode-codicons/tree/master/src/icons. You can find these icons, and it's better to convert SVG to PNG first. Also, remember delete your own screenshots.